ai/anthropic/v0/main.go

```diff
@@ -0,0 +1,180 @@
+	}
+
+	// A simple if statement would be enough in a component with a single task.
+	// If the number of task grows, here is where the execution task would be
+	// selected.
+	switch task {
+	case textGenerationTask:
+		e.execute = e.generateText
+	default:
+		return nil, fmt.Errorf("unsupported task")
+	}
+	return &base.ExecutionWrapper{Execution: e}, nil
+}
+
+func (e *execution) Execute(_ context.Context, inputs []*structpb.Struct) ([]*structpb.Struct, error) {
+	outputs := make([]*structpb.Struct, len(inputs))
+
+	// An execution  might take several inputs. One result will be returned for
+	// each one of them, containing the execution output for that set of
+	// parameters.
+	for i, input := range inputs {
+		output, err := e.execute(input)
+		if err != nil {
+			return nil, err
+		}
+
+		outputs[i] = output
+	}
+
+	return outputs, nil
+}
+
+func retriveMessageContent(contentsPbValue *structpb.ListValue) []content {
+	contents := []content{}
+	for _, contentPbValue := range contentsPbValue.GetValues() {
+		contentType := contentPbValue.GetStructValue().Fields["type"].GetStringValue()
+		// anthrothpic models does not support image urls
+		if contentType == "text" {
+			content := content{
+				Type: "text",
+				Text: contentPbValue.GetStructValue().Fields["text"].GetStringValue(),
+			}
+			contents = append(contents, content)
+		}
+	}
+	return contents
+}
+
+func retriveChatMessage(chatHistoryPbList *structpb.ListValue) []message {
+	messages := []message{}
+
+	for _, messagePbValue := range chatHistoryPbList.GetValues() {
+		contents := retriveMessageContent(messagePbValue.GetStructValue().Fields["content"].GetListValue())
+		completeMessage := message{Role: messagePbValue.GetStructValue().Fields["role"].GetStringValue(), Content: contents}
+		messages = append(messages, completeMessage)
+	}
+	return messages
+}
+
+func (e *execution) generateText(in *structpb.Struct) (*structpb.Struct, error) {
+	client := newClient(e.Setup, e.GetLogger())
+
+	prompt := in.Fields["prompt"].GetStringValue()
+
+	messages := []message{}
+
+	chatHistory := in.Fields["chat_history"].GetListValue()
+
+	if chatHistory != nil {
+		messages = retriveChatMessage(chatHistory)
+	}
+
+	finalMessage := message{
+		Role:    "user",
+		Content: []content{{Type: "text", Text: prompt}},
+	}
+
+	if in.Fields["prompt_images"] != nil {
+		for _, image := range in.Fields["prompt_images"].GetListValue().GetValues() {
+			// need to add support for different file types in the future.
+			image := content{
+				Type:   "image",
+				Source: &source{Type: "base64", MediaType: "image/jpeg", Data: base.TrimBase64Mime(image.GetStringValue())},
+			}
+			finalMessage.Content = append(finalMessage.Content, image)
+		}
+	}
+
+	messages = append(messages, finalMessage)
+
+	body := messagesReq{
+		Messages:    messages,
+		Model:       in.Fields["model_name"].GetStringValue(),
```

Comment:
Important
please remember to change to kebab-case when you modify the schema.

Reply:
fixed